crates/top/re_sdk/src/grpc_server.rs

```diff
@@ -0,0 +1,79 @@
+use re_log_types::LogMsg;
+
+/// A [`crate::sink::LogSink`] tied to a hosted Rerun gRPC server.
+///
+/// The hosted gRPC server may be connected to by any SDK or Viewer.
+///
+/// All data sent through this sink is immediately redirected to the gRPC server.
+pub struct GrpcServerSink {
+    /// Sender to send messages to the gRPC server.
+    sender: re_smart_channel::Sender<LogMsg>,
+
+    /// The gRPC server thread.
+    _server_handle: std::thread::JoinHandle<()>,
+
+    /// Rerun websocket server.
+    server_shutdown_signal: re_grpc_server::shutdown::Signal,
+}
+
+impl GrpcServerSink {
+    /// A `bind_ip` of `"0.0.0.0"` is a good default.
+    pub fn new(
+        bind_ip: &str,
+        grpc_port: u16,
+        server_memory_limit: re_memory::MemoryLimit,
+    ) -> Result<Self, std::net::AddrParseError> {
+        let (server_shutdown_signal, shutdown) = re_grpc_server::shutdown::shutdown();
+
+        let grpc_server_addr = format!("{bind_ip}:{grpc_port}").parse()?;
+        let (channel_tx, channel_rx) = re_smart_channel::smart_channel::<re_log_types::LogMsg>(
+            re_smart_channel::SmartMessageSource::MessageProxy {
+                url: format!("rerun+http://{grpc_server_addr}/proxy"),
+            },
+            re_smart_channel::SmartChannelSource::Sdk,
+        );
+        let server_handle = std::thread::Builder::new()
+            .name("message_proxy_server".to_owned())
+            .spawn(move || {
+                let mut builder = tokio::runtime::Builder::new_current_thread();
+                builder.enable_all();
+                let rt = builder.build().expect("failed to build tokio runtime");
+
+                rt.block_on(re_grpc_server::serve_from_channel(
```

Comment:
Will this work even if the user already has `#[tokio::main]` on their `fn main`?

Reply:
Yes, we're spawning a new thread, so anything goes.